velox/type/DecimalUtil.cpp

```diff
@@ -89,4 +89,28 @@ int32_t DecimalUtil::toByteArray(int128_t value, char* out) {
   return length;
 }
 
+void DecimalUtil::computeAverage(
+    int128_t& avg,
+    const int128_t& sum,
+    int64_t count,
+    int64_t overflow) {
+  if (overflow == 0) {
+    divideWithRoundUp<int128_t, int128_t, int64_t>(
+        avg, sum, count, false, 0, 0);
+  } else {
+    __uint128_t quotMul, remMul;
```

Comment:
We are emulating `int256` arithmetic here. 
In the issue, I proposed to use the boost multi-precision library to avoid reimplementing.
But this works too.

Reply:
Since we already implemented it all the way to final aggregation, there is no point to introduce new dependency to fix this in a 10 liner function.

Reply:
Thanks for clarifying!